app/domain/batch_process.rb

```diff
@@ -0,0 +1,17 @@
+class BatchProcess
+  attr_reader :output_documents
+
+  def initialize(output_documents)
+    @output_documents = output_documents
+  end
+
+  def csv
+    [headers].concat(output_documents.map(&:csv)).join
```

Comment:
I'm not quite sure about this. We are trusting (we know) that `output_document.csv` is implemented using `OutputDocument::CSVRenderer`.

If `output_document` returned a hash we could keep all knowledge of how to generate a CSV private to this class and also use the standard library to generate the CSV.